os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c

```diff
@@ -1185,6 +1185,24 @@ int rtw_printf_info(const char *format,...)
 	return ret;
 }
 
+/* Blocks the currently running task, need to called within the currently running task.
+ * It is added because there is no task suspension related function found in the SDK.
+ */
+void rtw_block_current_task(void){
+       FAR struct tcb_s *tcb;
+       tcb = this_task();
+       up_block_task(tcb, TSTATE_WAIT_FIN);
+}
+
+/* Unblocks the task previously blocked by rtw_block_current_task(), the input *task
+ * is the pid of the previously blocked task.
+ */
+void rtw_unblock_task(void *task){
+       FAR struct tcb_s *tcb;
+       tcb = sched_gettcb(*(pid_t *)task);
```

Comment:
Could you check that `tcb` is NULL or not?
Or how about adding assert if `tcb` cannot be NULL?
Because `sched_gettcb` can return NULL if `task` is already terminated.